### PR TITLE
Citadel: Flatten the most common aggregations 

### PR DESCRIPTION
**Pull request by plourenco**

### CODE REVIEW

File: citadel/indico_citadel/util.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def format_filters(params, filters, range_filters):
     :param params: The filter query params
     :param filters: The filter whitelist
     :param range_filters: The range filter whitelist
-    :returns: filters, extracted placeholders
+    :returns filters, extracted placeholders
```
Comment (ThiefMaster) [inline citadel/indico_citadel/util.py:107]: I don't think this is correct; sphinx uses `:returns: whatever` and `:return: whatever` (AFAIK we use the second version more frequently)

File: citadel/indico_citadel/util.py [hunk 2]
```diff
@@ -136,3 +136,36 @@ def remove_none_entries(obj):
...
+
+def format_aggregations(aggregations, filters):
+    """Format aggregations into a bucket dictionary.
+
+    Besides transforming each aggregation, ensures each bucket key
+    contains the most common result as defined from Elastic Search.
+
+    :param aggregations: The raw aggregation object
+    :param filters: The filter whitelist
+    :returns key: {label, buckets: [{key, count}]}
```
Comment (ThiefMaster) [inline citadel/indico_citadel/util.py:149]:
```suggestion
    :return: ``key: {label, buckets: [{key, count}]}`` dict
```

File: citadel/indico_citadel/util.py [hunk 3]
```diff
@@ -136,3 +136,36 @@ def remove_none_entries(obj):
...
+                'count': bucket['docCount']
+            } for bucket in value['buckets']]
+        }
+        for key, value in _flatten(aggregations)
+        if key in filters
+    }
+
+
+def _flatten(obj, target_key='buckets', parent_key=''):
+    if type(obj) is not dict:
```
Comment (ThiefMaster) [inline citadel/indico_citadel/util.py:166]:
```suggestion
    if not isinstance(obj, dict):
```

File: citadel/indico_citadel/util.py (hunk 1 above)
Comment (plourenco) [reply citadel/indico_citadel/util.py]: Thanks, we had different formats and I had the impression you would point it out if I didn't choose the right one 🙂

**Pull request merged**